src/tally.F90

```diff
@@ -383,6 +385,124 @@ score = wgt * mu ! avoid function call overhead
             j = j + (t % moment_order(j) + 1)**2 - 1
             cycle SCORE_LOOP
 
+          case (SCORE_NDPP_SCATT_N)
+            ! Skip any event where the particle didn't scatter
+            if (p % event /= EVENT_SCATTER) cycle SCORE_LOOP
+
+            ! For the case of analog NDPP scatter-n tallying, filter_index needs
+            ! to be adjusted to point to the first energyout filter
+            call tally_ndpp_n(p % event_nuclide, &
+              matching_bins(t % find_filter(FILTER_ENERGYIN)), &
+              score_index, filter_index - &
+              matching_bins(t % find_filter(FILTER_ENERGYOUT)) + 1, &
+              t % moment_order(j), last_wgt, .True., p % last_E, t % results)
+
+            cycle SCORE_LOOP
+
+          case (SCORE_NDPP_SCATT_PN)
+            ! Skip any event where the particle didn't scatter
+            if (p % event == EVENT_SCATTER) then
+              ! For the case of analog NDPP scatter-pn tallying, filter_index needs
+              ! to be adjusted to point to the first energyout filter
+              call tally_ndpp_pn(p % event_nuclide, &
+                matching_bins(t % find_filter(FILTER_ENERGYIN)), &
+                score_index, filter_index - &
+                matching_bins(t % find_filter(FILTER_ENERGYOUT)) + 1, &
+                t % moment_order(j), last_wgt, .True., p % last_E, t % results)
+            end if
+
+            j = j + t % moment_order(j)
+            cycle SCORE_LOOP
+
+          case (SCORE_NDPP_SCATT_YN)
+            ! Skip any event where the particle didn't scatter
+            if (p % event == EVENT_SCATTER) then
+              ! For the case of analog NDPP scatter-pn tallying, filter_index needs
+              ! to be adjusted to point to the first energyout filter
+              call tally_ndpp_yn(p % event_nuclide, &
+                matching_bins(t % find_filter(FILTER_ENERGYIN)), &
+                score_index, filter_index - &
+                matching_bins(t % find_filter(FILTER_ENERGYOUT)) + 1, &
+                t % moment_order(j), last_wgt, .True., p % last_E, &
+                p % last_uvw, t % results)
+            end if
+
+            j = j + (t % moment_order(j) + 1)**2 - 1
+            cycle SCORE_LOOP
+
+          case (SCORE_NDPP_NU_SCATT_N)
+            ! Skip any event where the particle didn't scatter
+            if (p % event /= EVENT_SCATTER) cycle SCORE_LOOP
+
+            ! For the case of analog NDPP nu-scatter-n tallying, filter_index needs
+            ! to be adjusted to point to the first energyout filter
+            call tally_ndpp_n(p % event_nuclide, &
+              matching_bins(t % find_filter(FILTER_ENERGYIN)), &
+              score_index, filter_index - &
+              matching_bins(t % find_filter(FILTER_ENERGYOUT)) + 1, &
+              t % moment_order(j), last_wgt, .True., p % last_E, t % results, .True.)
```

Comment:
We should use the post-collision weight rather than the pre-collision weight for nu-scatter, right?